test/gurka/gurka.cc

```diff
@@ -571,6 +571,41 @@ findEdgeByNodes(valhalla::baldr::GraphReader& reader,
                            end_node_name);
 }
 
+/**
+ * Finds a node in the graph based on its node name
+ *
+ * @param reader           graph reader to look up tiles and edges
+ * @param nodes            a lookup table from node names to coordinates
+ * @param node_name        name of the node
+ * @return the node_id
+ */
+baldr::GraphId findNode(valhalla::baldr::GraphReader& reader,
+                        const nodelayout& nodes,
+                        const std::string& node_name) {
+  // Iterate over all the tiles, there wont be many in unit tests..
+  for (auto tile_id : reader.GetTileSet()) {
+    auto tile = reader.GetGraphTile(tile_id);
+    // Iterate over all directed edges to find one with the name we want
+    for (const auto& e : tile->GetDirectedEdges()) {
+      // Check end node
+      auto end_node_id = e.endnode();
+      auto ll = reader.GetGraphTile(end_node_id)->get_node_ll(end_node_id);
+      if (ll.ApproximatelyEqual(nodes.at(node_name))) {
+        return end_node_id;
+      }
+      // Check begin node
+      auto edge_id = tile_id;
+      edge_id.set_id(&e - tile->directededge(0));
+      auto begin_node_id = reader.edge_startnode(edge_id);
+      ll = tile->get_node_ll(begin_node_id);
+      if (ll.ApproximatelyEqual(nodes.at(node_name))) {
+        return begin_node_id;
+      }
+    }
```

Comment:
you can just check the nodes directly:

```suggestion
    auto node_id = tile_id;
    for (auto node_id = tile_id; node_id.id() < tile->header()->nodecount(); ++node_id) {
      auto ll = tile->get_node_ll(node_id);
      if (ll.ApproximatelyEqual(nodes.at(node_name))) {
        return node_id;
      }
    }
```